### ISSUE TITLE
Randomly Crashing without Error

### ISSUE THREAD

**Issue by tt1884457** [OpenBMB/VoxCPM#300]
### Environment

* OS: Windows 11
* GPU: RTX 3060 12GB
* funasr version: 1.3.1
* Python 3.10.9

### Description

The application appears to freeze or terminate after generating some audios.
After some progress output, the console stops and eventually returns:

```text
Drücken Sie eine beliebige Taste . . .
```

The issue happens consistently on my system.

### Relevant Logs

```text
funasr version: 1.3.1

Running on device: cuda, dtype: bfloat16
Warning: torch.compile disabled - triton is not installed

Loaded VoxCPM2Model

Warm up VoxCPMModel...

100%|██████████████████████████████████████████████████████████████████████████████████| 10/10 [00:02<00:00, 3.93it/s]

17%|█████████████▍                                                                   | 77/466 [00:17<01:28, 4.41it/s]
17%|██████████████                                                                   | 81/466 [00:19<01:31, 4.20it/s]
15%|████████████▏                                                                    | 70/466 [00:16<01:33, 4.23it/s]

Drücken Sie eine beliebige Taste . . .
```

### Additional Notes

* CUDA is detected correctly.
* The model loads successfully before the freeze/crash.
* `torch.compile` is disabled because Triton is not installed.
* The issue occurs during the warmup/inference stage, not during model download.

### Question

Is this a known issue on Windows or related to:

* Triton not being installed?
* `bfloat16` support on RTX 3060?
* VoxCPM warmup/inference on Windows?

Any help or debugging suggestions would be appreciated.

Full Log:
```
funasr version: 1.3.1.
Downloading Model from https://www.modelscope.cn to directory: C:\Users\alexa\.cache\modelscope\hub\models\iic\SenseVoiceSmall
WARNING:root:trust_remote_code: False
* Running on local URL:  http://0.0.0.0:8808
* To create a public link, set `share=True` in `launch()`.
rtf_avg: 0.046: 100%|████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  2.19it/s]
Fetching 9 files: 100%|██████████████████████████████████████████████████████████████████████████| 9/9 [00:00<?, ?it/s]
Download complete: : 0.00B [00:00, ?B/s]              voxcpm_model_path: C:\Users\alexa\.cache\huggingface\hub\models--openbmb--VoxCPM2\snapshots\bffb3df5a29440629464e5e839f4d214c8714c3d, zipenhancer_model_path: iic/speech_zipenhancer_ans_multiloss_16k_base, enable_denoiser: True
Download complete: : 0.00B [00:00, ?B/s]
Loading AudioVAE from pytorch: C:\Users\alexa\.cache\huggingface\hub\models--openbmb--VoxCPM2\snapshots\bffb3df5a29440629464e5e839f4d214c8714c3d\audiovae.pth
Running on device: cuda, dtype: bfloat16
Loading model from safetensors: C:\Users\alexa\.cache\huggingface\hub\models--openbmb--VoxCPM2\snapshots\bffb3df5a29440629464e5e839f4d214c8714c3d\model.safetensors
Warning: torch.compile disabled - triton is not installed
Loaded VoxCPM2Model
Downloading Model from https://www.modelscope.cn to directory: C:\Users\alexa\.cache\modelscope\hub\models\iic\speech_zipenhancer_ans_multiloss_16k_base
2026-05-10 20:27:09,750 - modelscope - INFO - initiate model from C:\Users\alexa\.cache\modelscope\hub\models\iic\speech_zipenhancer_ans_multiloss_16k_base
2026-05-10 20:27:09,751 - modelscope - INFO - initiate model from location C:\Users\alexa\.cache\modelscope\hub\models\iic\speech_zipenhancer_ans_multiloss_16k_base.
2026-05-10 20:27:09,753 - modelscope - INFO - initialize model from C:\Users\alexa\.cache\modelscope\hub\models\iic\speech_zipenhancer_ans_multiloss_16k_base
2026-05-10 20:27:09,979 - modelscope - WARNING - No preprocessor field found in cfg.
2026-05-10 20:27:09,980 - modelscope - WARNING - No val key and type key found in preprocessor domain of configuration.json file.
2026-05-10 20:27:09,980 - modelscope - WARNING - Cannot find available config to build preprocessor at mode inference, current config: {'model_dir': 'C:\\Users\\alexa\\.cache\\modelscope\\hub\\models\\iic\\speech_zipenhancer_ans_multiloss_16k_base'}. trying to build by task and model information.
2026-05-10 20:27:09,981 - modelscope - INFO - No preprocessor key ('speech_zipenhancer_ans_multiloss_16k_base', 'acoustic-noise-suppression') found in PREPROCESSOR_MAP, skip building preprocessor. If the pipeline runs normally, please ignore this log.
Warm up VoxCPMModel...
100%|██████████████████████████████████████████████████████████████████████████████████| 10/10 [00:02<00:00,  3.93it/s]
 17%|█████████████▍                                                                   | 77/466 [00:17<01:28,  4.41it/s]
 17%|██████████████                                                                   | 81/466 [00:19<01:31,  4.20it/s]
 15%|████████████▏                                                                    | 70/466 [00:16<01:33,  4.23it/s]
Drücken Sie eine beliebige Taste . . .
```

**Comment by liuxin99** [comment]
Thanks for the report. One thing I noticed is the last line:

Drücken Sie eine beliebige Taste . . .

This is the German Windows message for "Press any key to continue....". It is usually printed by cmd when a pause command is executed, for example at the end of a .bat / .cmd launcher. It is not printed by VoxCPM, Gradio, PyTorch, or Python itself.

Could you share the exact command or launcher you used to start the demo? For example, did you run the official command python app.py --port 8808, or did you start it through a .bat / .cmd file or another launcher?

**Comment by tt1884457** [comment]
I use a start.bat:

```
@echo off
call venv\Scripts\activate

python app.py

if errorlevel 1 (
    echo.
    echo Es ist ein Fehler aufgetreten.
)

pause
```

I also ran it with 
```
D:\AI\VoxCPM>venv\Scripts\activate

(venv) D:\AI\VoxCPM>python app.py --port 8808
```
But it crashes too

**Comment by liuxin99** [comment]
Thanks for confirming — the “Press any key…” line is just from `pause` in your `.bat`, not from VoxCPM. Since `python app.py` also crashes, we can ignore that.

From your log, Triton / `torch.compile` and bf16 are unlikely the cause (compile is already disabled, and warm-up succeeded). The silent exit with no Python traceback usually means a hard CUDA / process abort.

On RTX 3060 12GB this is often **VRAM pressure**: `app.py` may load VoxCPM2 + ZipEnhancer (+ ASR if Ultimate Cloning is on), which can be tight and crash on later generations.

Please try:

```bat
voxcpm --text "Hello world" -o out.wav --no-denoiser --no-optimize
```

If that stays stable, also check GPU memory before the crash, and try shorter reference audio / target text in the demo.

Could you share whether the CLI command above survives repeated runs, and the GPU memory usage right before `app.py` dies?

**Comment by tt1884457** [comment]
I generated 5 audios with the cli command, everything worked. Then I generated around 11 audios with the UI. It crashed with 92% memory usage.
```
100%|██████████████████████████████████████████████████████████████████████████████| 10/10 [00:05<00:00,  1.74it/s]
 32%|█████████████████████████▏                                                     | 7/22 [00:03<00:08,  1.79it/s]
 14%|██████████▊                                                                    | 3/22 [00:01<00:11,  1.60it/s]
 36%|████████████████████████████▋                                                  | 8/22 [00:04<00:07,  1.86it/s]
 18%|██████████████▎                                                                | 4/22 [00:02<00:10,  1.70it/s]
 32%|█████████████████████████▏                                                     | 7/22 [00:03<00:08,  1.80it/s]
 73%|████████████████████████████████████████████████████████▋                     | 16/22 [00:08<00:03,  1.91it/s]
  Badcase detected, audio_text_ratio=8.5, retrying...
 41%|████████████████████████████████▎                                              | 9/22 [00:04<00:06,  1.91it/s]
 32%|█████████████████████████▏                                                     | 7/22 [00:03<00:08,  1.80it/s]
 27%|█████████████████████▌                                                         | 6/22 [00:03<00:09,  1.76it/s]
100%|██████████████████████████████████████████████████████████████████████████████| 22/22 [00:10<00:00,  2.00it/s]
  Badcase detected, audio_text_ratio=11.0, retrying...
 45%|███████████████████████████████████▍                                          | 10/22 [00:04<00:05,  2.03it/s]
 55%|██████████████████████████████████████████▌                                   | 12/22 [00:06<00:05,  1.90it/s]
  Badcase detected, audio_text_ratio=6.5, retrying...
 45%|███████████████████████████████████▍                                          | 10/22 [00:05<00:06,  1.91it/s]
```

**Comment by liuxin99** [comment]
Thanks, that helps a lot.

CLI being stable while the UI crashes at ~92% memory strongly points to **VRAM pressure in the Gradio demo** (VoxCPM2 + ZipEnhancer, and possibly ASR), not Triton/bf16.

Also, the `Badcase detected ... retrying...` lines mean the same request can rerun inference up to 3 times, which increases peak memory usage.

Workarounds for now on 12GB:
- Prefer CLI with `--no-denoiser` when possible
- In the UI, keep reference audio / target text short, and avoid Ultimate Cloning / denoise if not needed
